perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/FacilityAttributeProcessor.java

```diff
@@ -21,69 +21,91 @@
-				log.error("Error setting attribute:", e);
+		} catch (NamingException | InternalErrorException e) {
+			log.error("Error setting attribute:", e);
 		}
 	}
 
 	public void processAttributeRemoved(String msg, MessageBeans beans) {
-		Perun perun = ldapcManager.getPerunBl();
+		PerunBl perun = (PerunBl) ldapcManager.getPerunBl();
 
 		// ensure we have the correct beans available
-		if(beans.getAttributeDef() == null || beans.getFacility() == null) {
+		if (beans.getAttributeDef() == null || beans.getFacility() == null) {
 			return;
 		}
 		try {
 			log.debug("Getting resources assigned to facility {}", beans.getFacility().getId());
 			// List<Resource> resources = Rpc.FacilitiesManager.getAssignedResources(ldapcManager.getRpcCaller(), beans.getFacility());
-			List<Resource> resources = perun.getFacilitiesManager().getAssignedResources(ldapcManager.getPerunSession(), beans.getFacility());
-			for(Resource resource: resources) {
+			List<Resource> resources = perun.getFacilitiesManagerBl().getAssignedResources(ldapcManager.getPerunSession(), beans.getFacility());
+			for (Resource resource : resources) {
 				log.debug("Removing attribute {} from resource {}", beans.getAttributeDef(), resource);
 				perunResource.modifyEntry(resource, beans.getAttributeDef());
 			}
-		} catch (NamingException | FacilityNotExistsException | InternalErrorException | PrivilegeException e) {
+		} catch (NamingException | InternalErrorException e) {
 			log.error("Error removing attribute:", e);
 		}
 	}
 
 	public void processAllAttributesRemoved(String msg, MessageBeans beans) {
-		Perun perun = ldapcManager.getPerunBl();
+		PerunBl perun = (PerunBl) ldapcManager.getPerunBl();
 
 		// ensure we have the correct beans available
-		if(beans.getUser() == null) {
+		if (beans.getFacility() == null) {
 			return;
 		}
 		try {
 			log.debug("Getting resources assigned to facility {}", beans.getFacility().getId());
 			// List<Resource> resources = Rpc.FacilitiesManager.getAssignedResources(ldapcManager.getRpcCaller(), beans.getFacility());
-			List<Resource> resources = perun.getFacilitiesManager().getAssignedResources(ldapcManager.getPerunSession(), beans.getFacility());
-			for(Resource resource: resources) {
+			List<Resource> resources = perun.getFacilitiesManagerBl().getAssignedResources(ldapcManager.getPerunSession(), beans.getFacility());
+			for (Resource resource : resources) {
 				log.debug("Removing all attributes from resource {}", resource);
 				perunResource.removeAllAttributes(resource);
 			}
-		} catch (NamingException | FacilityNotExistsException | InternalErrorException | PrivilegeException e) {
+		} catch (NamingException | InternalErrorException e) {
 			log.error("Error removing attributes:", e);
 		}
 	}
 
-}
+	public void processVirtualAttributeChanged(String msg, MessageBeans beans) {
+		PerunBl perun = (PerunBl) ldapcManager.getPerunBl();
+		if (beans.getAttribute() == null || beans.getFacility() == null) {
+			return;
+		}
+		try {
```

Comment:
Can you please rebase this pull-request ? Please adjust implementation to update attributes also on recently added ``perunFacility`` and not just assigned resources. Thank you.